feat(workspace): add a new resource workspace eip associate (#3885)
huaweicloud/terraform-provider-huaweicloud: docs/resources/workspace_eip_associate.md

---
subcategory: "Workspace"
---

# huaweicloud_workspace_eip_associate

> Associates an EIP to a specified desktop.

Reviewer: ```suggestion
Using this resource to associate an EIP to a specified desktop within HuaweiCloud.
```

Author: fixed

## Example Usage

### Associate with an EIP ID

```hcl
variable "desktop_id" {}
variable "eip_id" {}

resource "huaweicloud_workspace_eip_associate" "test" {
  desktop_id = var.desktop_id
  eip_id     = var.eip_id
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to associate the EIP. If omitted, the provider-level
  region will be used. Changing this will create a new resource.

* `desktop_id` - (Required, String, ForceNew) Specifies the desktop ID. Changing this will create a new resource.

* `eip_id` - (Required, String, ForceNew) Specifies the EIP ID to associate. Changing this will create a new resource.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

> * `id` - The associate EIP in UUID format.

Reviewer: ```suggestion
* `id` - The resource ID, the format is `<desktop_id>/<eip_id>`.
```

Author: fixed

> * `enterprise_project_id` - The enterprise project ID to which associate the EIP.

Reviewer: ```suggestion
* `enterprise_project_id` - The enterprise project ID to which the EIP associated.
```

Author: fixed

* `public_ip` - The address of the EIP.

## Import

> EIP association can be imported using the `id` of the EIP, e.g.

Reviewer: ```suggestion
EIP association can be imported using the `desktop_id` and associated `eip_id`, separated by a slash, e.g.
```

Author: fixed
